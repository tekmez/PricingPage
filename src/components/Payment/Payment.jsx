import { Text, View } from 'react-native'
import ShieldIcon from '../../../assets/svg/shieldIcon.svg'
import styles from './Payment.style'
const Payment = () => {
  return (
    <View testID="payment-container" style={styles.container}>
      <ShieldIcon width={16} height={16} />
      <Text testID="payment-title" style={styles.text}>
        NO PAYMENT NOW
      </Text>
    </View>
  )
}

export default Payment
