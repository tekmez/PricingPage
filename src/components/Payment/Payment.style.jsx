import { StyleSheet } from 'react-native'

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    width: 134,
    height: 16,
    gap: 4,
    alignSelf: 'center',
    marginTop: 6,
  },
  text: {
    fontSize: 12,
    lineHeight: 16,
    color: '#717578',
    fontFamily: 'Inter-Medium',
  },
})

export default styles
