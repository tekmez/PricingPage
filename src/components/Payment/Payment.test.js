import React from 'react'
import { render } from '@testing-library/react-native'
import Payment from './Payment'
jest.mock('../../../assets/svg/shieldIcon.svg', () => () => (
  <div testID="shield-icon" />
))
describe('Payment', () => {
  it('should match snapshot', () => {
    const comp = render(<Payment />).toJSON()
    expect(comp).toMatchSnapshot()
  })

  it('should render correctly', () => {
    const comp = render(<Payment />)
    const paymentTitle = comp.getByTestId('payment-title').children[0]
    expect(paymentTitle).toBe('NO PAYMENT NOW')
  })
})
