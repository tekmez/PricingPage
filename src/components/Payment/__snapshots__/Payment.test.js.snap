// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payment should match snapshot 1`] = `
<View
  style={
    {
      "alignSelf": "center",
      "flexDirection": "row",
      "gap": 4,
      "height": 16,
      "marginTop": 6,
      "width": 134,
    }
  }
  testID="payment-container"
>
  <div
    testID="shield-icon"
  />
  <Text
    style={
      {
        "color": "#717578",
        "fontFamily": "Inter-Medium",
        "fontSize": 12,
        "lineHeight": 16,
      }
    }
    testID="payment-title"
  >
    NO PAYMENT NOW
  </Text>
</View>
`;
